sdk/identity/Azure.Identity/tests/TestEnvVar.cs

```diff
@@ -2,29 +2,82 @@
 // Licensed under the MIT License.
 
 using System;
-using System.Collections;
 using System.Collections.Generic;
-using System.Text;
+using System.Threading;
 
 namespace Azure.Identity.Tests
 {
     internal class TestEnvVar : IDisposable
     {
-        private readonly string _origValue = null;
-        private readonly string _name;
+        private static SemaphoreSlim _lock = new SemaphoreSlim(1, 1);
+        private readonly Dictionary<string, string> _originalValues = new Dictionary<string, string>()
+        {
+            { "AZURE_USERNAME", EnvironmentVariables.Username},
```

Comment:
Why do you need to use EnvironmentVariables.* when you already have a variable name here?

Reply:
I wanted to make it clear that these variables are represented in `EnvironmentVariables`. Ideally I wanted to do this with reflection, to always keep in sync with the variables represented there, but for that to work, I'd have to also rename the properties to match the environment variable names.